rust/src/lightning.rs

```diff
@@ -583,20 +599,9 @@ pub fn run_ldk(system: &LightningSystem) -> Result<BackgroundProcessor> {
 
     INVOICE_PAYER.set(invoice_payer.clone());
 
-    // Step 19: Background Processing
-    let background_processor = BackgroundProcessor::start(
-        system.persister.clone(),
-        invoice_payer.clone(),
-        system.chain_monitor.clone(),
-        system.channel_manager.clone(),
-        GossipSync::p2p(system.gossip_sync.clone()),
-        system.peer_manager.clone(),
-        system.logger.clone(),
-        Some(scorer),
-    );
+    let background_processor = system.start_background_processor();
```

Comment:
🔧 I'm not a huge fan of calling this here (inside of `run_ldk`) and also on the outside when restarting the background processor.

Actually, this function appears to be doing only 2 things:

- Setting the `INVOICE_PAYER` static variable, something that should only happen once during the lifetime of the app AFAIK.
- Starting the background processor via the `LightningSystem`.

As such, I think we should get rid of the `run_ldk` function and just do those two things separately when necessary.

Reply:
agreed, but 

> Some more refactoring could be probably done there, but I tried to keep the changes to a minimum.